src/common/json_request.cpp

```
@@ -57,6 +62,15 @@ void JsonRequest::ParseCommon(const std::string& request_body) {
     if (auto parsed_value = SafeGetFromJsonPath(json_body, {"data_batch", "datatype"})) {
         datatype_ = *parsed_value;
     }
+    if (auto parsed_value = SafeGetFromJsonPath(json_body, {"data_batch", "datatype_length"})) {
+        // Parse as integer since datatype_length is optional
+        try {
+            datatype_length_ = std::stoi(*parsed_value);
+        } catch (const std::exception&) {
+            // If parsing fails, leave datatype_length_ as std::nullopt
+            datatype_length_ = std::nullopt;
```

this feel like a type of silent failure - where the value may have been populated in  the json struct, but was not an int (e.g. it was a string). It does make sense to default to `nullopt` if no value was in the json, but can we differentiate between a non-existing value vs an incorrectly-typed one?